File: Native_iOS/FinanzApp/Models/Scheduling.swift
import Foundation

enum Scheduling {
    static func getNextDueDate(currentDueDate: String, frequency: String) -> String {
        guard var date = Date(yyyyMMDD: currentDueDate) else { return currentDueDate }
        switch frequency {
        case "daily":  date = Calendar.current.date(byAdding: .day, value: 1, to: date) ?? date
        case "weekly": date = Calendar.current.date(byAdding: .day, value: 7, to: date) ?? date
        case "monthly": date = Calendar.current.date(byAdding: .month, value: 1, to: date) ?? date
        case "yearly": date = Calendar.current.date(byAdding: .year, value: 1, to: date) ?? date
        default: break
        }
        let f = DateFormatter()
        f.dateFormat = "yyyy-MM-dd"
        f.locale = Locale(identifier: "en_US_POSIX")
        return f.string(from: date)
    }

    static func isDue(_ nextDueDate: String) -> Bool {
        return nextDueDate <= Date.currentString
    }

    static func frequencyLabel(_ frequency: String) -> String {
        switch frequency {
        case "daily": return "Giornaliero"
        case "weekly": return "Settimanale"
        case "monthly": return "Mensile"
        case "yearly": return "Annuale"
        default: return frequency
        }
    }
}

File: src/utils/scheduling.ts
export function getNextDueDate(
  currentDueDate: string,
  frequency: 'daily' | 'weekly' | 'monthly' | 'yearly'
): string {
  const date = new Date(currentDueDate + 'T00:00:00');

  switch (frequency) {
    case 'daily':
      date.setDate(date.getDate() + 1);
      break;
    case 'weekly':
      date.setDate(date.getDate() + 7);
      break;
    case 'monthly':
      date.setMonth(date.getMonth() + 1);
      break;
    case 'yearly':
      date.setFullYear(date.getFullYear() + 1);
      break;
  }

  const year = date.getFullYear();
  const month = String(date.getMonth() + 1).padStart(2, '0');
  const day = String(date.getDate()).padStart(2, '0');
  return `${year}-${month}-${day}`;
}

export function isDue(nextDueDate: string): boolean {
  const today = new Date();
  const todayStr = `${today.getFullYear()}-${String(today.getMonth() + 1).padStart(2, '0')}-${String(today.getDate()).padStart(2, '0')}`;
  return nextDueDate <= todayStr;
}

export function getFrequencyLabel(frequency: string): string {
  switch (frequency) {
    case 'daily': return 'Giornaliero';
    case 'weekly': return 'Settimanale';
    case 'monthly': return 'Mensile';
    case 'yearly': return 'Annuale';
    default: return frequency;
  }
}
